awx/main/tasks.py

```diff
@@ -224,11 +315,15 @@ def cluster_node_heartbeat(self):
             instance_list.remove(inst)
     if this_inst:
         startup_event = this_inst.is_lost(ref_time=nowtime)
-        if this_inst.capacity == 0:
+        if this_inst.capacity == 0 and this_inst.enabled:
             logger.warning('Rejoining the cluster as instance {}.'.format(this_inst.hostname))
-        this_inst.capacity = get_system_task_capacity()
-        this_inst.version = awx_application_version
-        this_inst.save(update_fields=['capacity', 'version', 'modified'])
+        if this_inst.enabled:
+            this_inst.refresh_capacity()
+            handle_ha_toplogy_changes.apply_async()
```

Comment:
(1) We can run `handle_ha_topology_changes` inline both here and below. (2) Do we need to run this every heartbeat?